x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/aws_credentials_form/aws_input_var_fields.tsx

```diff
@@ -6,39 +6,76 @@
  */
 
 import React from 'react';
-import { EuiFieldPassword, EuiFieldText, EuiFormRow } from '@elastic/eui';
+import { EuiFieldText, EuiFormRow, EuiSpacer } from '@elastic/eui';
+import { PackageInfo } from '@kbn/fleet-plugin/common';
+import { css } from '@emotion/react';
+import { PackagePolicyInputVarField } from '../package_policy_input_var_field';
 import { AwsOptions } from './get_aws_credentials_form_options';
+import { findVariableDef } from '../utils';
 
 export const AwsInputVarFields = ({
   fields,
   onChange,
+  packageInfo,
 }: {
   fields: Array<AwsOptions[keyof AwsOptions]['fields'][number] & { value: string; id: string }>;
   onChange: (key: string, value: string) => void;
-}) => (
-  <div>
-    {fields.map((field) => (
-      <EuiFormRow key={field.id} label={field.label} fullWidth hasChildLabel={true} id={field.id}>
+  packageInfo: PackageInfo;
+}) => {
+  return (
+    <div>
+      {fields.map((field) => (
         <>
-          {field.type === 'password' && (
-            <EuiFieldPassword
-              id={field.id}
-              type="dual"
-              fullWidth
-              value={field.value || ''}
-              onChange={(event) => onChange(field.id, event.target.value)}
-            />
+          {field.type === 'password' && field.isSecret === true && (
+            <>
+              <EuiSpacer size="m" />
+              <div
+                css={css`
+                  width: 100%;
+                  .euiFormControlLayout,
+                  .euiFormControlLayout__childrenWrapper,
+                  .euiFormRow,
+                  input {
+                    max-width: 100%;
+                    width: 100%;
+                  }
+                `}
+              >
+                <PackagePolicyInputVarField
+                  varDef={{
+                    ...findVariableDef(packageInfo, field.id)!,
```

Comment:
so the thing with the component 'PackagePolicyInputVarField' is that I added this as a **duplicate** from the Fleet team's folder. The reason on why I don't just import it from the Fleet's team folder is that when I added export statement on Fleet team's index.ts so that we can import it, the CI complains about Page load bundle size goes over the limit.
By making this duplicate and only exporting+importing required types, function, component instead, we minimize the Page load bundle size

Reply:
tbh, I have concerns about duplicating the component into our codebase. First of all, we won't be able to benefit from all the changes that the fleet team would make. A good example is exactly the secrets implementation. We couldn't leverage the secrets implemented by Fleet because we've built our components instead of using what Fleet builds and provides. With any change or bugfix, we will need to backport them into our component. what most likely would happen is that we won't see all the changes to `PackagePolicyInputVarField` in Fleet and these components just diverge over time. I think we should check with the Fleet team if we could use the component from Fleet before deciding to duplicate it. @kpollich any input in regards to exporting `PackagePolicyInputVarField` from Fleet and using it in other plugins? What would be the right approach?

Reply:
Is the page load bundle size limit warning happening in cloud_security_posture or fleet?

Have we had a problem with tree shaking in Kibana?

Reply:
Exporting the component from Fleet and using a lazy import in other plugins is the best path forward here to share the code. It avoids the bundle size issues, and the component is small enough that loading it on-demand via the lazy import should have no meaningful impact on page load time or responsiveness.